tests/bgp/test_startup_tsa_tsb_service.py

```diff
@@ -278,10 +247,12 @@ def test_tsa_tsb_service_with_dut_cold_reboot(duthosts, localhost, enum_rand_one
                 pytest.fail("Not all ipv6 routes are announced to neighbors")
 
     finally:
-
-        # Verify DUT is in normal state.
-        pytest_assert(TS_NORMAL == get_traffic_shift_state(duthost),
-                      "DUT is not in normal state")
+        # Bring back the supervisor and line cards to the BGP operational normal state
+        initial_tsa_check_before_and_after_test(duthosts)
+        # Verify DUT is in normal state after cold reboot scenario.
+        if not (int_status_result and crit_process_check and TS_NORMAL == get_traffic_shift_state(duthost)):
+            logging.info("DUT is not in normal state after cold reboot, doing config-reload")
+            config_reload(duthost, safe_reload=True, check_intf_up_ports=True)
```

Comment:
Would it be good to indicate what caused the DUT to not be normal state?

Reply:
Added a clear logger statement indicating details.